unit_test/tstUniformCartesianMeshMapping.hpp

```diff
@@ -0,0 +1,347 @@
+
+    EXPECT_EQ( global_mesh.lowCorner( 0 ), 0.0 );
+    EXPECT_EQ( global_mesh.lowCorner( 1 ), 0.0 );
+    EXPECT_EQ( global_mesh.lowCorner( 2 ), 0.0 );
+
+    EXPECT_EQ( global_mesh.highCorner( 0 ), num_cell );
+    EXPECT_EQ( global_mesh.highCorner( 1 ), num_cell );
+    EXPECT_EQ( global_mesh.highCorner( 2 ), num_cell );
+
+    EXPECT_EQ( global_grid.globalNumEntity( Cajita::Cell(), 0 ), num_cell );
+    EXPECT_EQ( global_grid.globalNumEntity( Cajita::Cell(), 1 ), num_cell );
+    EXPECT_EQ( global_grid.globalNumEntity( Cajita::Cell(), 2 ), num_cell );
+
+    EXPECT_TRUE( global_grid.isPeriodic( 0 ) );
+    EXPECT_FALSE( global_grid.isPeriodic( 1 ) );
+    EXPECT_TRUE( global_grid.isPeriodic( 2 ) );
+
+    auto local_grid = manager->mesh().localGrid();
+    EXPECT_EQ( local_grid->haloCellWidth(), 1 );
+
+    // Check mapping.
+    auto ghosted_cells = local_grid->indexSpace(
+        Cajita::Ghost(), Cajita::Cell(), Cajita::Local() );
+    auto local_mesh = Cajita::createLocalMesh<TEST_MEMSPACE>( *local_grid );
+    Kokkos::View<double*** [3], TEST_MEMSPACE> cell_forward_map(
+        "cell_forward_map", ghosted_cells.extent( Dim::I ),
+        ghosted_cells.extent( Dim::J ), ghosted_cells.extent( Dim::K ) );
+    Kokkos::View<double*** [3], TEST_MEMSPACE> cell_reverse_map(
+        "cell_reverse_map", ghosted_cells.extent( Dim::I ),
+        ghosted_cells.extent( Dim::J ), ghosted_cells.extent( Dim::K ) );
+    Kokkos::View<bool***, TEST_MEMSPACE> cell_map_success(
+        "cell_reverse_map_success", ghosted_cells.extent( Dim::I ),
+        ghosted_cells.extent( Dim::J ), ghosted_cells.extent( Dim::K ) );
+    Kokkos::View<double*** [3], TEST_MEMSPACE> default_cell_reverse_map(
+        "default_cell_reverse_map", ghosted_cells.extent( Dim::I ),
+        ghosted_cells.extent( Dim::J ), ghosted_cells.extent( Dim::K ) );
+    Kokkos::View<bool***, TEST_MEMSPACE> default_cell_map_success(
+        "default_cell_reverse_map_success", ghosted_cells.extent( Dim::I ),
+        ghosted_cells.extent( Dim::J ), ghosted_cells.extent( Dim::K ) );
+    Cajita::grid_parallel_for(
+        "check_mapping", TEST_EXECSPACE{}, ghosted_cells,
+        KOKKOS_LAMBDA( const int i, const int j, const int k ) {
+            // Map to physical frame.
+            LinearAlgebra::Vector<double, 3> cell_coords;
+            int ijk[3] = { i, j, k };
+            local_mesh.coordinates( Cajita::Cell{}, ijk, cell_coords.data() );
+            LinearAlgebra::VectorView<double, 3> phys_coords(
+                &cell_forward_map( i, j, k, 0 ), cell_forward_map.stride( 3 ) );
+            CurvilinearMeshMapping<mapping_type>::mapToPhysicalFrame(
+                mapping, cell_coords, phys_coords );
+
+            // Map to reference frame.
+            LinearAlgebra::VectorView<double, 3> ref_coords(
+                &cell_reverse_map( i, j, k, 0 ), cell_reverse_map.stride( 3 ) );
+            ref_coords = { cell_coords( Dim::I ) - 0.1,
```

Comment:
do we need this here?

Reply:
Good question. The container (i.e. the `Kokkos::View`) is const but the data that it points to is not. Note the the definition of all `operator()` in a `Kokkos::View` is actually `const` to allow for inlining.